src/app/_validators/sitting-records-form-validator.ts

```diff
@@ -0,0 +1,107 @@
+import { AbstractControl, FormGroup, Validators } from '@angular/forms';
+
+// setup simple regex for white listed characters
+const numbersOnly = new RegExp('^[0-9]+$');
+
+
+// create your class that extends the angular validator class
+export class CustomValidators extends Validators {
+
+  static validateDateFormat(control: FormGroup) {
+    if(control){
+        const { dateDay, dateMonth, dateYear } = control.controls
+
+        if(dateDay.touched && (dateDay.value == '')){
+            return { 'date_invalid' : true}
+        }
+        if(dateMonth.touched && (dateMonth.value == '')){
+            return { 'date_invalid' : true}
+        }
+        if(dateYear.touched && (dateYear.value == '' || dateYear.value.length != 4)){
+            return { 'date_invalid' : true}
+        }
+
+        if(dateDay.value && dateMonth.value && (dateYear.value && dateYear.value.length == 4)){
+            const dateObj = new Date(`${dateMonth.value}-${dateDay.value}-${dateYear.value}`)
+            const todaysDate = new Date()
+            const supportedDate = new Date('12-11-2021')
+
+            if(dateObj > todaysDate){
+                return { 'date_after_today': true }
+            }
+
+            if(dateObj < supportedDate){
+                return { 'date_out_of_support': true }
+            }
+
+            const validDay = CustomValidators.validateDay(dateDay)
+            const validMonth = CustomValidators.validateMonth(dateMonth)
+            const validYear = CustomValidators.validateYear(dateYear)
+
+            if(validDay != true){
+              return { 'date_invalid' : true }
+            }
+            if(validMonth != true){
+              return { 'date_invalid' : true } 
+            } 
+            if(validYear != true){
+              return { 'date_invalid' : true } 
+            }
```

Comment:
Can we remove this section and add check for validateDay, validateMonth, validateYear in the above section of date_invalid by adding OR statement

Reply:
function updated to remove above functions and perform checks in single lines